### PR TITLE
Updated outdated dependencies

### PR DESCRIPTION
**Pull request by Abdulwahaab710**
### In this PR I have updated the following outdated dependencies:
- http
- rake
- recursive-open-struct
- rubocop
- webmock

### CODE REVIEW

**Comment by abonas** [comment]
hi, @Abdulwahaab710 .
please squash your commits, they have a similar name, and they have the same purpose.
also, travis is failing, please check its error messages. mostly I believe it's due to version bump of rubocop and the fix of [this](https://github.com/bbatsov/rubocop/issues/4039).

**Comment by Abdulwahaab710** [comment]
Done :)

**Comment by simon3z** [comment]
@moolitayer can you review? (mostly for new dependencies versions implications)

**Comment by cben** [comment]
- dev dependencies are safe to bump.
- https://github.com/aetherknight/recursive-open-struct/blob/master/CHANGELOG.md
1.0.3 adds ruby 2.4 compat, but stops worrying about 2.0, 2.1 which kubeclient still supports.  However 2.0, 2.1 still pass tests.  I hope they'll bump to at least 1.1.0 when it actually breaks 2.0, 2.1.
- https://github.com/httprb/http/blob/master/CHANGES.md looks sane.

**Comment by moolitayer** [comment]
> I hope they'll bump to at least 1.1.0 when it actually breaks ruby 2.0...

I doubt if they will even notice when that happens...

We probably want to enjoy @jrafanie's https://github.com/aetherknight/recursive-open-struct/pull/52 with all the performance work going around, so I would consider dropping support of ruby 2.0 & 2.1 and bumping kubeclient's version. I'm not sure how/if that is problematic to users.

@kirs @grosser @albertrdixon care to comment on dropping ruby 2.0, 2.1 support ?

**Comment by grosser** [comment]
everything up to 2.1 is unsupported https://www.ruby-lang.org/en/news/2017/04/01/support-of-ruby-2-1-has-ended/ so I'm in favor of getting rid of it ... make the new gem have a `require_ruby_version` and it should not break for people stuck on older versions ...

**Comment by jrafanie** [comment]
> We probably want to enjoy @jrafanie's aetherknight/recursive-open-struct#52 with all the performance work going around,

@moolitayer I doubt that would help most use cases... I would think you would need an enormous recursive structure to see much difference.  I could be wrong though. 

I'm all for dropping unsupported rubies in a non-patch release 👍

**Comment by grosser** [comment]
FYI ROS removal issue https://github.com/abonas/kubeclient/issues/251

**Comment by kirs** [comment]
👍 for dropping anything < 2.1

**Comment by simon3z** [comment]
@Abdulwahaab710 @grosser @moolitayer @jrafanie we may want to change the `.travis.yml` as well at some point (now is running against `2.0` and `2.1` as well).

**Comment by simon3z** [comment]
@Abdulwahaab710 if you squash your 2 commits in 1 I think we're good to merge.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,5 @@
 language: ruby
 rvm:
-  - "2.0"
-  - "2.1"
```
Comment (Abdulwahaab710) [inline .travis.yml]: @simon3z I have removed rvm 2.0 and rvm 2.1

**Comment by cben** [comment]
We forgot to update `required_ruby_version` in gemspec, it's still 2.0.0 but we now have non-2.0 syntax.

**Pull request merged**